internal/js_library/js_library.bzl

```diff
@@ -73,6 +73,11 @@ If package_path is set, the linker will link this package under <package_path>/n
 If package_path is not set the this will be the root node_modules of the workspace.""",
     ),
     "srcs": attr.label_list(allow_files = True),
+    "source_directory": attr.bool(
```

Comment:
wdyt about making this private/internal and only use it from `npm_install`? One less thing to support etc 🤷